vm/src/obj/objsequence.rs

```diff
@@ -16,12 +16,12 @@ use super::objlist::PyList;
 use super::objslice::{PySlice, PySliceRef};
 use super::objtuple::PyTuple;
 
-pub trait PySliceableSequence {
-    fn do_slice(&self, range: Range<usize>) -> Self;
-    fn do_slice_reverse(&self, range: Range<usize>) -> Self;
-    fn do_stepped_slice(&self, range: Range<usize>, step: usize) -> Self;
-    fn do_stepped_slice_reverse(&self, range: Range<usize>, step: usize) -> Self;
-    fn empty() -> Self;
+pub trait PySliceableSequence<S: PySliceableSequence<S>> {
+    fn do_slice(&self, range: Range<usize>) -> S;
```

Comment:
Maybe have the output type be an associated type rather than a generic?

Reply:
Fixed it